src/test/java/de/tum/in/www1/artemis/exercise/programmingexercise/ProgrammingExerciseTestService.java

```diff
@@ -1416,11 +1418,15 @@ void exportProgrammingExerciseInstructorMaterial_shouldReturnFile(boolean saveEm
             assertThat(listOfIncludedFiles).anyMatch((filename) -> filename.toString().matches(".*-exercise.zip"))
                     .anyMatch((filename) -> filename.toString().matches(".*-solution.zip")).anyMatch((filename) -> filename.toString().matches(".*-tests.zip"))
                     .anyMatch((filename) -> filename.toString().matches(EXPORTED_EXERCISE_PROBLEM_STATEMENT_FILE_PREFIX + ".*.md"))
-                    .anyMatch((filename) -> filename.toString().matches(EXPORTED_EXERCISE_DETAILS_FILE_PREFIX + ".*.json"));
+                    .anyMatch((filename) -> filename.toString().matches(EXPORTED_EXERCISE_DETAILS_FILE_PREFIX + ".*.json"))
+                    .anyMatch((filename) -> BUILD_PLAN_FILE_NAME.equals(filename.toString()));
             if (saveEmbeddedFiles) {
                 assertThat(listOfIncludedFiles).anyMatch((filename) -> filename.toString().equals(embeddedFileName1))
                         .anyMatch((filename) -> filename.toString().equals(embeddedFileName2));
             }
+            Path buildPlanPath = listOfIncludedFiles.stream().filter(file -> BUILD_PLAN_FILE_NAME.equals(file.getFileName().toString())).findFirst().orElseThrow();
+            String buildPlanContent = Files.readString(extractedZipDir.resolve(buildPlanPath), StandardCharsets.UTF_8);
+            assertThat(buildPlanContent).isEqualTo("my build plan");
```

Comment:
The method is called from both the `ProgrammingExerciseBitbucketBambooIntegrationTest` and its GitlabJenkins counterpart. Where does the distinction happen if the build plan should be there or not? I would expected the build plan file to be missing in the Bamboo case since that feature is not supported there.

Reply:
i adjusted the test now that it only adds a build plan to the exercise if it is a gitlab/jenkins setup. because initially I just always added it to the exercise ignoring the setup.